### PR TITLE
safe tools client: Add cancel payment option to the options menu in scheduled payment card

### PR DESCRIPTION
**Pull request by jurgenwerk**
This PR adds an option to cancel a payment.

Options menu:
image

Confirmation modal:
image

Canceling:
image

Success:
image

Failure:
image

### CODE REVIEW

File: packages/boxel/addon/components/boxel/menu/index.css [hunk 1]
```diff
@@ -29,6 +29,10 @@
   border-radius: 0 0 var(--boxel-border-radius) var(--boxel-border-radius);
 }
 
+.boxel-menu__item:only-child {
```
Comment (jurgenwerk) [inline packages/boxel/addon/components/boxel/menu/index.css:32]: Before:

image

After: 
image

File: packages/cardpay-sdk/sdk/scheduled-payment-module.ts [hunk 2]
```diff
@@ -838,10 +838,12 @@ export default class ScheduledPaymentModule {
     });
   }
 
-  async cancelScheduledPayment(scheduledPaymentId: string, hubRootUrl?: string) {
+  async cancelScheduledPayment(scheduledPaymentId: string, hubRootUrl?: string, authToken?: string) {
     let hubAuth = await getSDK('HubAuth', this.ethersProvider, hubRootUrl, this.signer);
     hubRootUrl = await hubAuth.getHubUrl();
-    let authToken = await hubAuth.authenticate();
+    if (!authToken) {
```
Comment (jurgenwerk) [inline packages/cardpay-sdk/sdk/scheduled-payment-module.ts:844]: Do not prompt to authenticate when already authenticated. It is expected the user is authenticated at this point when this is called.

**Comment by jurgenwerk** [comment]
There is something left to be desired - reloading the future payments list after canceling so that the canceled payment isn't shown anymore. Will add this in another PR.

**Review by alex-cardstack — APPROVED** [review]

**Review by FadhlanR — APPROVED** [review]

File: packages/safe-tools-client/app/components/future-payments-list/scheduled-payment-card/index.gts [hunk 3]
```diff
@@ -41,21 +105,100 @@ export default class ScheduledPaymentCard extends Component<Signature> {
...
+                <FailureIcon class='action-chin-info-icon' />
+                <span>{{this.cancelationErrorMessage}}</span>
+              </a.InfoArea>
+            {{else if this.paymentCanceled}}
+              <a.ActionButton {{on "click" this.closeCancelScheduledPaymentModal}} data-test-close-cancel-payment-modal>
+                Close
+              </a.ActionButton>
+
+              <a.InfoArea>
+                <SuccessIcon class='action-chin-info-icon' />
```
Comment (lukemelia) [inline packages/safe-tools-client/app/components/future-payments-list/scheduled-payment-card/index.gts:175]: Since this seems like a common pattern, perhaps the InfoArea component should yield SuccessIcon and FailureIcon components with the class pre-applied. OK to do this in a separate PR.

**Review by lukemelia — APPROVED** [review]

File: packages/safe-tools-client/app/components/future-payments-list/scheduled-payment-card/index.gts (hunk 3 above)
Comment (jurgenwerk) [reply packages/safe-tools-client/app/components/future-payments-list/scheduled-payment-card/index.gts:175]: Yeah I agree. I added a ticket for that.
